basil/HL/binder_mk56.py

```diff
@@ -0,0 +1,110 @@
+#
+# ------------------------------------------------------------
+# Copyright (c) All rights reserved
+# SiLab, Institute of Physics, University of Bonn
+# ------------------------------------------------------------
+#
+
+import struct
+
+from basil.HL.binder_mk53 import binderMK53
+
+# MODBus function codes
+FUNCTION_READN = 0x03  # read n words
+FUNCTION_READN_ALT = 0x04  # read n words, does not occur
+FUNCTION_WRITE = 0x06  # write word
+FUNCTION_WRITEN = 0x10  # write n words
+
+# Error codes
+ERROR_CODES = {
+    1: "Invalid function",
+    2: "Invalid parameter address",
+    3: "Pparameter value outside range of values",
+    4: "Slave not ready",
+    5: "Write access to parameter denied"
+}
+
+
+class binderMK56(binderMK53):
+
+    '''Driver for the Binder MK 53 climate chamber.
+    A protocoll _similar_ to MODBus via RS 422 serial port is used with 9600 baud rate.
+    Credits to ecree-solarflare with some further information at https://github.com/ecree-solarflare/ovenctl
+    '''
+
+    def __init__(self, intf, conf):
+        super(binderMK56, self).__init__(intf, conf)
+
+    def init(self):
+        super(binderMK56, self).init()
+        self.ADDR_CURTEMP = 0x1004
+        self.ADDR_SETPOINT = 0x10b2
+        self.ADDR_MANSETPT = 0x114c
+
+        self.slave_address = self._init['address']  # set the device address
+        self.min_temp = self._init['min_temp']  # define the minimum temperature one can set, for safety
+        self.max_temp = self._init['max_temp']  # define the maximum temperature one can set, for safety
+
+    def get_door_open(self):
+        raise NotImplementedError("Get Door not available for MK56.")
+
+    def get_mode(self):
+        raise NotImplementedError("Get Mode not available for MK56.")
+
+    def set_temperature(self, temperature, reps=10):
+        if temperature < self.min_temp:
+            raise RuntimeWarning('Set temperature %f is lower than minimum allowed temperature %f' % (temperature, self.min_temp))
+        if temperature > self.max_temp:
+            raise RuntimeWarning('Set temperature %f is higher than maximum allowed temperature %f' % (temperature, self.max_temp))
+        for _ in range(reps):
+            try:
+                self.write(self.ADDR_MANSETPT, self._encode_float(temperature))
+                break
+            except RuntimeWarning:
+                pass
```

Comment:
The exceptions raised here should be ValueErrors not RuntimeWarnings. If you only want to warn and not crash I suggest you use the `warnings` module.

Reply:
Also the `%` style formatting is deprecated, nowadays use f-strings or the format-Method